### PR TITLE
Production workloads on gemstash

### PR DESCRIPTION
**Pull request by JackDanger**

### CODE REVIEW

**Comment by JackDanger** [comment]
cc: @phanle

File: docker-compose.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ services:
       DATABASE_URL: postgres://postgres@db
       CACHE_EXPIRATION: 3600
       CACHE_MAX_SIZE: 25000
+      PUMA_WORKERS: 8
```
Comment (JackDanger) [inline docker-compose.yml]: Previously the default was 0 workers (disabling cluster mode, the thing that lets Puma behave like Unicorn) and 16 threads. This gets us 8x throughput and Unicorn semantics.

Comment (phanle) [reply docker-compose.yml]: We might not want to overload 1. The boxes itself. 2. The DB.
I spotted some spikes in those metrics yesterday 
https://app.datadoghq.com/dash/994564/gemstash-2-cloned?live=false&page=0&is_auto=false&from_ts=1542670838641&to_ts=1542678823753&tile_size=m&fullscreen=false
screen shot 2018-11-20 at 8 44 50 am

 I would suggest we do 2 workers (or 4) first and observe the behaviors before bumping to 8.

Comment (JackDanger) [reply docker-compose.yml]: I've got a separate PR to make the boxes bigger. Great point about the db though, let's make sure that can handle the load.

Comment (genericbeckham) [reply docker-compose.yml]: What's the worry with overloading the host and/or the db? Isn't this just a case where we don't want to refuse a connection? Let the bajillions of hosts that run bundle every deploy make a connection and wait their turn. They only fail when they can't get that connection, right?

**Comment by phanle** [comment]
@JackDanger Let me know when you landed and deployed this. We could do some performance testing

**Comment by phanle** [comment]
Great work @JackDanger. I am glad that we could stabilize our deployments with this change.

**Pull request merged**